app/uk/gov/hmrc/hec/controllers/SDESCallbackController.scala

```diff
@@ -0,0 +1,128 @@
+        )
+        callbackNotification.notification match {
+          case FileReady | FileReceived => Future.successful(Ok)
+          case FileProcessed            =>
+            deleteFileAndUpdateTaxChecks(callbackNotification.filename, callbackNotification.correlationID)
+          case FileProcessingFailure    =>
+            deleteFileFromObjectStore(callbackNotification.filename)
+        }
+
+      case JsError(err) =>
+        logger.warn(s"Failed to parse the SDES callback notification with error:: $err")
+        Future.successful(BadRequest)
+    }
+
+  }
+
+  private def deleteFileFromObjectStore(fileName: String)(implicit
+    hc: HeaderCarrier,
+    hecTaxCheckExtractionContext: HECTaxCheckExtractionContext
+  ) = {
+    val dirName = getDirName(fileName)
+    fileStoreService
+      .deleteFile(fileName, dirName)
+      .fold(
+        { err =>
+          logger.warn(s"Failed to delete file:: $fileName from object store with error: $err")
+          InternalServerError
+        },
+        _ => Ok
+      )
+  }
+
+  private def deleteFileAndUpdateTaxChecks(fileName: String, correlationId: String)(implicit
+    hc: HeaderCarrier,
+    hecTaxCheckExtractionContext: HECTaxCheckExtractionContext
+  ) = {
+    val dirName               = getDirName(fileName)
+    val deleteAndUpdateResult =
+      for {
+        _ <- fileStoreService.deleteFile(fileName, dirName)
+        _ <- if (dirName === taxCheckDir) updateHecTaxCheck(correlationId) else EitherT.pure[Future, models.Error](())
+      } yield ()
+
+    deleteAndUpdateResult.fold(
+      { err =>
+        logger.warn(s"Failure in deleting file :: $fileName and update the hec tax check list with error :: $err")
+        InternalServerError
+      },
+      _ => Ok
+    )
+  }
+
+  private def updateHecTaxCheck(correlationId: String)(implicit
+    hc: HeaderCarrier,
+    hecTaxCheckExtractionContext: HECTaxCheckExtractionContext
+  ) = for {
+    hecTaxChecks          <-
+      taxCheckService.getAllTaxCheckCodesByCorrelationId(correlationId)
+    updatedHecTaxCheckList = hecTaxChecks.map(_.copy(fileCorrelationId = None, isExtracted = true))
+    _                     <- taxCheckService.updateAllHecTaxCheck(updatedHecTaxCheckList)
+  } yield ()
+
+  // object store delete need dir name to delete the file but
+  //call back from SDES doesn't hold this information
+  // So, Identifying the directory name of the file from the map
+  //which has  data like partial file name ->  dir name
+  private def getDirName(filename: String) = {
+    val keys: Set[String] = FileMapOps.fileNameDirMap.keySet
+    val result            = keys.filter(pf =>
+      filename.contains(pf)
+    ) //filter map keys, if the file name(from SDES) contains the partial file name from map key
+    FileMapOps.getDirName(
+      result.headOption.getOrElse(sys.error("File doesn't belong to hec"))
```

Comment:
I think you can shorten all of this to something like:
```scala
FileMapOps.fileNameDirMap
  .find{ case (key, _) => fileName.contains(key) }
  .map(_._2)
  .getOrElse(sys.error(...))
```

Reply:
yup nicer this way